suprb/optimizer/solution/saga/crossover.py

```diff
@@ -0,0 +1,82 @@
+import numpy as np
+from abc import ABCMeta, abstractmethod
+from suprb.base import BaseComponent
+from .solution_extension import SagaSolution, NPoint, Uniform
+from suprb.utils import RandomState
+
+
+class SagaCrossover(BaseComponent):
```

Comment:
isnt this effectively still duplicating ga.crossover?

Reply:
Yes and no. For the `SagaCrossover` we use a `SagaSolution` which is necessary because the `SagaSolution` has more member parameters than `Solution`. I will try to make it common, perhaps with a Union